sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.h

```diff
@@ -100,6 +100,7 @@ typedef struct _nib_onl_entry {
      * @note    Only available if @ref CONFIG_GNRC_IPV6_NIB_QUEUE_PKT != 0.
      */
     gnrc_pktqueue_t *pktqueue;
+    size_t pktqueue_len;                       /**< Number of queued packets */
```

Comment:
We only need this in the error path, so no need to cache this information.
Let's save some memory and do

```C
static inline size_t gnrc_pktqueue_len(const gnrc_pktqueue_t *queue)
{
    size_t len = 0;

    while (queue->next) {
        queue = queue->next;
        ++len;
    }

    return len;
}
```

Reply:
What about a large neighbor base (~128) and a high-bandwidth link (e.g. ethernet)? This will get executed for each packet you want to put on the wire.

Reply:
Hm ok but it's only the sending to a neighbor that is already unreachable that would then be slow, right?
(Not sure if we can speed up the loop much by only considering neighbors in `GNRC_IPV6_NIB_NC_INFO_NUD_STATE_PROBE`) 

The alternative with the 128 neighbor NIB would be an additional 512 bytes of RAM for the counting.

Reply:
The whole network stack will be slow, I'm generally trying to avoid an unpredictable slowdown.

> The alternative with the 128 neighbor NIB would be an additional 512 bytes of RAM for the counting.

I aggre that's a lot. I made it `uint8_t` and moved it at the struct's end where 1 byte is wasted anyway because of padding.

The more I dig through the network stack I realize it is not really made for these numbers.